react-web/src/pages/userProfile/UserProfile.tsx

```diff
@@ -74,129 +125,314 @@ const UserProfile = () => {
+      }).catch((errorObj) => {
+        let errorMsg = 'Something went wrong. Please try again.'
+        if (errorObj?.response?.data) {
+          errorMsg = errorObj.response.statusText + ' - ' + errorObj.response.data 
+        }
+        setShowError(errorMsg);
+        const timeout = setTimeout(() => { clearTimeout(timeout); setShowError("") }, 5000)
+        
       })
     };
     submitProfile();
   };
 
+  useEffect(() => {
+    if (focussedOwnerRepo) {    
+      if (timer) {
+        clearTimeout(timer)
+        setTimer(null);
+        dispatch(hideConfirmConnection());
+      }
+      const newTimer = setTimeout(() => {
+        if (owner.length && repo.length) {
+          clearOwnerRepoError()
+          setCanShowConnectModal(true)
+          dispatch(verifyRepoAccess({owner: owner, repo: repo}));
+        } else {
+          isOwnerRepoValidationError()
+        }
+      }, 600)
+      setTimer(newTimer)
+    }
+  }, [owner, repo])
+
+  const inputChanged = (e: any,type:string) => {
+    const currentVal = e.target.value
+    if(type === 'owner') {
+      setOwner(currentVal)
+    } else if (type ==='repo') {
+      setRepo(currentVal)
+    }
+    setCanShowConnectModal(false)
+    setFocussedOwnerRepo(true)
+    dispatch(hideConfirmConnection());
+  }
+
+  const inputBlurred = (e: any, type: string) => {
+    setFocussedOwnerRepo(false)
+    setCanShowConnectModal(false)
+  }
+  
+  const CLIENT_ID = "10c45fe6a101a3abe3a0"; //of CertificationApp OAuthApp owned by CertGroup in anusree91
+  const CLIENT_SECRET = "9fe0553a4727c47b434e732c3e258ba54c992051";
+  // Lists all repo and have to manually click 'Grant' for the repo we need. 
+  // If that is not chosen, it doesn't ask again for permission
+
+
+  // const CLIENT_ID = "Iv1.78d108e12af627c4" // of CertTestApp GithubApp under CertGroup in anusree91
+  // const CLIENT_SECRET = "a77d34c1ffde5c4a7f0749fd7f969d6bbcda1ae7"
+
+  useEffect(() => {
+    // to extract ?code= from the app base url redirected after Github connect
+    if (githubAccessCode) {
+      (async () => {
+        const params = "?client_id=" + CLIENT_ID + "&client_secret=" + CLIENT_SECRET + "&code=" + githubAccessCode
+        const access_token = await dispatch(getUserAccessToken({queryParams: params}))
+        searchParams.delete("code");
+        setSearchParams(searchParams);
+        const formData = form.getValues()
+        setCanShowConnectModal(false)
+        setFocussedOwnerRepo(false)
+        const timeout = setTimeout(async () => {
+          clearTimeout(timeout);
+          setCanShowConnectModal(true)
+          clearTimeout(timer)
+          setTimer(null)
+          await dispatch(verifyRepoAccess({owner: formData.owner, repo: formData.repo}))
+          localStorage.removeItem('profile')
+          localStorage.removeItem('accessToken')
+        }, 0)
+      })()
+    } else {
+      setCanShowConnectModal(false)
+      localStorage.removeItem('profile')
+    }
+  }, [])
```

Comment:
do we want this to be run only once? can we do the following:
- suppress the eslint warning for `react-hooks/exhaustive-deps` on this line
- add a comment and explain why is safe to run this effect only once

Reply:
suppressed the warning

Reply:
still, we lack an explanation why we run this only once